### PR TITLE
feat: Add earth2studio-create-diagnostic AI agent skill

### PR DESCRIPTION
**Pull request by NickGeneva**
## Summary

Adds a new AI agent skill for creating Earth2Studio diagnostic model wrappers. Diagnostic models perform single-step transformations (no time integration), such as computing derived variables, precipitation estimation, or probabilistic super-resolution.

### Skill Documentation (`skills/earth2studio-create-diagnostic/SKILL.md`)
- Complete workflow for implementing diagnostic models in Earth2Studio
- Covers three diagnostic model types:
  - **Simple** - Basic transformations with no checkpoints (e.g., wind speed from u/v components)
  - **AutoModel** - Models that load weights from packages (NGC/HuggingFace)
  - **Generative** - Probabilistic models producing multiple samples (e.g., CorrDiff-style diffusion)
- Step-by-step implementation guide with coordinate handling and forward pass patterns
- Clear differentiation from prognostic models (no `create_iterator`, no `PrognosticMixin`, no `lead_time` dimension)

### Reference Materials (`skills/earth2studio-create-diagnostic/references/`)
- `skeleton-template.py` - Full model skeleton with FILL comments for all three model types
- `testing-guide.py` - Test patterns with mock fixtures and package creation

### Evaluation Suite (`skills/earth2studio-create-diagnostic/evals/`)
- 3 evaluation scenarios covering all diagnostic model types:
  1. Simple diagnostic (WindSpeed from u10m/v10m)
  2. AutoModel diagnostic (PrecipEstimator with normalization buffers)
  3. Generative diagnostic (SuperResolution with sample dimension)
- Docker environment and bootstrap scripts for isolated evaluation
- Target implementations and tests for grading

## Key Patterns Documented

- `@batch_func` and `@batch_coords` decorator usage
- `handshake_dim`/`handshake_coords` for coordinate validation
- `register_buffer` for normalization parameters
- Sample dimension handling for generative models
- `check_optional_dependencies` for optional package imports

## Checklist

- [x] SPDX license headers included
- [x] Follows Earth2Studio skill structure
- [x] Evaluation suite with targets
- [x] Reference templates included

### CODE REVIEW

File: skills/earth2studio-create-diagnostic/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+
+**`load_model`:** Use `package.resolve()`, `map_location="cpu"`, `eval()` mode.
+
+```python
+@classmethod
+@check_optional_dependencies()
+def load_model(cls, package: Package) -> DiagnosticModel:
+    checkpoint_path = package.resolve("model.pt")
+    core_model = torch.load(checkpoint_path, map_location="cpu")
+    core_model.eval()
```
[the remark this answers is not in the document]
Comment (NickGeneva) [reply skills/earth2studio-create-diagnostic/SKILL.md]: Fixed. Replaced `@torch.inference_mode()` with `with torch.no_grad():` context manager inside `__call__` bodies (consistent with codebase convention) and added `weights_only=False` to the `torch.load` example.

File: skills/earth2studio-create-diagnostic/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,499 @@
...
+            "sample": np.arange(self.number_of_samples),
+            "variable": np.array(OUTPUT_VARS),
+            "lat": self.lat_output,
+            "lon": self.lon_output,
+        })
+
+    @batch_func()
+    def __call__(self, x, coords):
+        output_coords = self.output_coords(coords)
+        out = torch.zeros([len(v) for v in output_coords.values()], ...)
```
[the remark this answers is not in the document]
Comment (NickGeneva) [reply skills/earth2studio-create-diagnostic/SKILL.md]: Fixed. All templates now consistently use `with torch.no_grad():` context manager inside `__call__` instead of `@torch.inference_mode()` decorator.

File: skills/earth2studio-create-diagnostic/references/skeleton-template.py [hunk 3]
```diff
@@ -0,0 +1,635 @@
...
+    # =========================================================================
+    # 8. SINGLE-STEP FORWARD
+    # =========================================================================
+    @batch_func()
+    def __call__(
+        self,
+        x: torch.Tensor,
+        coords: CoordSystem,
+    ) -> tuple[torch.Tensor, CoordSystem]:
+        """Forward pass generating samples.
```
[the remark this answers is not in the document]
Comment (NickGeneva) [reply skills/earth2studio-create-diagnostic/references/skeleton-template.py]: Fixed. `GenerativeDiagnostic.__call__` now uses `with torch.no_grad():` consistent with the other templates.

File: skills/earth2studio-create-diagnostic/evals/targets/eval_3_target.py [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+
+        return out
+
+    @batch_func()
+    def __call__(
+        self,
+        x: torch.Tensor,
+        coords: CoordSystem,
+    ) -> tuple[torch.Tensor, CoordSystem]:
+        """Forward pass generating samples.
```
[the remark this answers is not in the document]
Comment (NickGeneva) [reply skills/earth2studio-create-diagnostic/evals/targets/eval_3_target.py:269]: Fixed. Added `with torch.no_grad():` to `SuperResolution.__call__`, consistent with the other eval targets.

File: skills/earth2studio-create-diagnostic/evals/targets/eval_2_target.py [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+    def load_default_package(cls) -> Package:
+        """Default pre-trained model package.
+
+        Returns
+        -------
+        Package
+            Model package with checkpoint files.
+        """
+        return Package(
+            "ngc://models/nvidia/modulus/precip_estimator@v1.0",
```
Comment (pzharrington) [inline skills/earth2studio-create-diagnostic/evals/targets/eval_2_target.py]: Just for my understanding, is this just sort of a placeholder/example fake package that the skill should reproduce? The test file just forces in a tmp package path so this never gets used, this is just for a sort of example reference? If so I guess it's not super relevant but maybe a hf package would be better since we prefer that now

File: skills/earth2studio-create-diagnostic/references/pr-body-template.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+Blocked by: OPTIONAL_PR_OR_ISSUE
+Closes: OPTIONAL_ISSUE
+
+### Diagnostic details
+
+| Property | Value |
+|---|---|
+| **Diagnostic type** | Simple derived / AutoModel / Generative |
+| **Architecture** | PyTorch / ONNX / JAX / other |
```
Comment (pzharrington) [inline skills/earth2studio-create-diagnostic/references/pr-body-template.md]: `Framework` instead of `Architecture`?

File: skills/earth2studio-create-diagnostic/references/validation-guide.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
+# Validation Guide for Diagnostic Model Wrappers
+
+This guide covers reference comparison, sanity plots, PR text, and review
+follow-up for diagnostic model wrappers.
+
+## Step 10 - Reference Comparison and Validation
```
Comment (pzharrington) [inline skills/earth2studio-create-diagnostic/references/validation-guide.md:6]: Step 10?

File: skills/earth2studio-create-diagnostic/SKILL.md [hunk 8]
```diff
@@ -0,0 +1,363 @@
...
+2. `input_coords`
+3. `output_coords` decorated with `@batch_coords()`
+4. `__str__` if useful
+5. `load_default_package` for AutoModel/generative diagnostics
+6. `load_model` for AutoModel/generative diagnostics
+7. `to` only when non-PyTorch state must move devices
+8. Private/support methods
+9. `__call__` decorated with `@torch.inference_mode()` and `@batch_func()`
+
+Do not create shared base classes or broad abstractions for one model.
```
Comment (pzharrington) [inline skills/earth2studio-create-diagnostic/SKILL.md]: Is this too restrictive? What if there's kinda multiple versions of the same model, e.g. a base DLESyM and a DLESyMLatLon subclass for convenience? Maybe more relevant for prognostics

File: skills/earth2studio-create-diagnostic/references/validation-guide.md (hunk 7 above)
Comment (pzharrington) [reply skills/earth2studio-create-diagnostic/references/validation-guide.md:6]: Ok now I see it's from the main `SKILL.md` order, I was reading the PR diff from top-to-bottom so saw this file before `SKILL.md`

File: skills/earth2studio-create-diagnostic/evals/targets/eval_2_target.py (hunk 5 above)
Comment (NickGeneva) [reply skills/earth2studio-create-diagnostic/evals/targets/eval_2_target.py]: good point, i agree we should switch to huggingface

File: skills/earth2studio-create-diagnostic/references/pr-body-template.md (hunk 6 above)
Comment (NickGeneva) [reply skills/earth2studio-create-diagnostic/references/pr-body-template.md]: Good call

File: skills/earth2studio-create-diagnostic/SKILL.md (hunk 8 above)
Comment (NickGeneva) [reply skills/earth2studio-create-diagnostic/SKILL.md]: Fair, I'll relax this slightly

Comment (NickGeneva) [reply skills/earth2studio-create-diagnostic/SKILL.md]: Updated!

File: skills/earth2studio-create-diagnostic/references/pr-body-template.md (hunk 6 above)
Comment (NickGeneva) [reply skills/earth2studio-create-diagnostic/references/pr-body-template.md]: Updated!

File: skills/earth2studio-create-diagnostic/evals/targets/eval_2_target.py (hunk 5 above)
Comment (NickGeneva) [reply skills/earth2studio-create-diagnostic/evals/targets/eval_2_target.py]: Updated!

**Review by pzharrington — APPROVED** [review]
LGTM
